### PR TITLE
fix: logger-not-working-in-plugins

### LINKED ISSUE #36073: [Bug]: Logger does not work from the /appsmith-plugins directory
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On adding `log.debug()`, `log.info()` in PostgresPlugin.java (and any other file in the /appsmith-plugins directory), these log statements are not printed to the console. 

https://github.com/appsmithorg/appsmith/blob/f02b448ded9bb29f3e6f18ceb4e970ef6423bab1/app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/PostgresPlugin.java#L386
https://github.com/appsmithorg/appsmith/blob/f02b448ded9bb29f3e6f18ceb4e970ef6423bab1/app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/PostgresPlugin.java#L658

### Steps To Reproduce

Configure the server locally and check for logs in the terminal after configuring postgres datasource.

### Public Sample App

_No response_

### Environment

Production

### Severity

Low (Cosmetic UI issues)

### Issue video log

_No response_

### Version

Appsmith 2d8892f3

### PR DESCRIPTION
**Pull request by AnnaHariprasad5123**
Fixes #36073 

Hi @NilanshBansal 

**Issue :**

**Missing Logging Implementation :**

- Without a logging implementation (such as SLF4J Simple or Logback) in the project's classpath, the logging statements in the plugins cannot be executed.
- As a result, no log output is being printed to the terminal.

**Solution :**

The solution is to add a logging implementation to the plugins parent pom. In this case, you can add the slf4j-simple dependency to your pom.xml file. This will provide a simple logging implementation that will output log statements to the console.

```
<dependency>
<groupId>org.slf4j</groupId>
<artifactId>slf4j-api</artifactId>
<version>1.7.36</version>
</dependency>

<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-simple</artifactId>
    <version>1.7.36</version> 
</dependency>
```
**Explanation:**

- slf4j-api provides the SLF4J API, which is the interface for logging.
  provides the SLF4J API, which is the interface for logging.
- slf4j-simple provides a simple implementation of the SLF4J API, which is responsible for actually printing the log messages to the console.

**Screenshots :** 

Amazon S3 Plugin and Postgres Plugin 
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging capabilities with the integration of SLF4J API and SLF4J Simple implementations.

- **Improvements**
	- Improved log management and output formatting for better monitoring and debugging across various plugins, transitioning from standard output to structured logging.
	- Refined logging practices in multiple plugins to support better maintainability and performance.
	- Removed method for console logging from the Stopwatch class to streamline logging practices.

### CODE REVIEW

**Comment by NilanshBansal** [comment]
@AnnaHariprasad5123 thanks for this fix!
Can you please also update the System.out.println() statements to log.debug() in the Plugins. 
This was recently changed in this PR https://github.com/appsmithorg/appsmith/pull/36077

**Comment by AnnaHariprasad5123** [comment]
Hi @NilanshBansal, I replaced all System.out.println to log.debug. Could you check once.

**Comment by AnnaHariprasad5123** [comment]
Hi @NilanshBansal, Could you check once. Let me know if any changes required.

**Comment by NilanshBansal** [comment]
@AnnaHariprasad5123 can you also remove the method stopAndLogTimeInMillisWithSysOut() from Stopwatch.java and also replace all the usages of this method with stopAndLogTimeInMillis(), since it was added as a temporary fix
https://github.com/appsmithorg/appsmith/pull/36077/files#diff-8005b0d844e9004cc03abebe5ec522c3ff8be090da2c9288344a5c2a76e4717b

image

**Comment by AnnaHariprasad5123** [comment]
Hi @NilanshBansal, I replaced all of them with with stopAndLogTimeInMillis(). Could you check now.

**Comment by NilanshBansal** [comment]
@AnnaHariprasad5123 I tried pulling and executing the changes locally, but I am not seeing some logs from PostgresPlugin.java.
For example this log is not getting printed.
https://github.com/appsmithorg/appsmith/blob/55f428c1b68c80ec1a714ade5657bd570487a4ec/app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/PostgresPlugin.java#L247
Can you check the reasons for this.

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 1]
```diff
@@ -923,15 +923,14 @@ public Mono<AmazonS3> datasourceCreate(DatasourceConfiguration datasourceConfigu
...
+            log.debug(printMessage);
             if (connection != null) {
                 Mono.fromCallable(() -> {
                             connection.shutdown();
                             return connection;
                         })
                         .onErrorResume(exception -> {
-                            System.out.println("Error closing S3 connection.");
-                            exception.printStackTrace();
+                            log.debug("Error closing S3 connection.", exception);
```
Comment (AnaghHegde) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:933]:
```suggestion
                            log.error("Error while closing S3 connection. ", exception.getMessage());
```

File: app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java [hunk 2]
```diff
@@ -108,8 +108,7 @@ public Mono<ActionExecutionResult> execute(
                         } catch (SQLException error) {
                             error.printStackTrace();
                             // This should not happen ideally.
-                            System.out.println(
-                                    "Error checking validity of Databricks connection : " + error.getMessage());
+                            log.debug("Error checking validity of Databricks connection : " + error.getMessage());
```
Comment (AnaghHegde) [inline app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java:111]:
```suggestion
                            log.error("Error checking validity of Databricks connection : " + error.getMessage());
```

File: app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java [hunk 3]
```diff
@@ -303,14 +301,14 @@ public Mono<Connection> datasourceCreate(DatasourceConfiguration datasourceConfi
...
-            System.out.println(printMessage);
+            log.debug(printMessage);
             try {
                 if (connection != null) {
                     connection.close();
                 }
             } catch (SQLException e) {
                 // This should not happen ideally.
-                System.out.println("Error closing Databricks connection : " + e.getMessage());
+                log.debug("Error closing Databricks connection : " + e.getMessage());
```
Comment (AnaghHegde) [inline app/server/appsmith-plugins/databricksPlugin/src/main/java/com/external/plugins/DatabricksPlugin.java:311]:
```suggestion
                log.error("Error closing Databricks connection : " + e.getMessage());
```

File: app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java [hunk 4]
```diff
@@ -952,12 +952,12 @@ public Mono<Firestore> datasourceCreate(DatasourceConfiguration datasourceConfig
...
         public Mono<DatasourceTestResult> testDatasource(DatasourceConfiguration datasourceConfiguration) {
             String printMessage = Thread.currentThread().getName() + ": testDatasource() called for Firestore plugin.";
-            System.out.println(printMessage);
+            log.debug(printMessage);
             return datasourceCreate(datasourceConfiguration).flatMap(connection -> {
                 try {
                     connection.listCollections();
                 } catch (FirestoreException e) {
-                    System.out.println("Invalid datasource configuration: " + e.getMessage());
+                    log.debug("Invalid datasource configuration: " + e.getMessage());
```
Comment (AnaghHegde) [inline app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:960]:
```suggestion
                    log.error("Invalid datasource configuration: " + e.getMessage());
```

Please user log.error when dealing with exception.

**Review by AnaghHegde — CHANGES_REQUESTED** [review]

**Comment by NilanshBansal** [comment]
@AnnaHariprasad5123 were you able to work on the changes suggested? And also fix the logging issue?

**Comment by AnnaHariprasad5123** [comment]
I changed all the logs. Now, I am checking the issue

**Comment by NilanshBansal** [comment]
> I changed all the logs. Now, I am checking the issue

Thank You, can you please commit the changes done until now, this will help us to review the rest of the things while you work on resolving the logger issue 🙏

**Comment by AnnaHariprasad5123** [comment]
Hi @NilanshBansal, log statements are visible in console now. I have added scope as provided for slf4j dependencies in parent pom of appsmith-plugins and removed slf4j exclusion from pf4j dependency.  I have checked some of the plugins with test- configurations action only. Could you check all logs in all plugins from your end.

**MySQL Plugin**
image
**Postgres Plugin**
image
**Amazon S3 Plugin** 
image
**Dynamo Plugin**
image

**Comment by NilanshBansal** [comment]
Thank You @AnnaHariprasad5123. I am checking this

**Review by NilanshBansal — APPROVED** [review]

**Comment by NilanshBansal** [comment]
Thanks @AnnaHariprasad5123 for your valuable contribution! We highly appreciate your efforts in making Appsmith better ❤️
You can find more issues to contribute from [this](https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Inviting+Contribution%22) list!

**Pull request merged**